src/umpire/alloc/SICMAllocator.hpp

```diff
@@ -0,0 +1,169 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_SICMAllocator_HPP
+#define UMPIRE_SICMAllocator_HPP
+
+#include <cstdlib>
+#include <list>
+#include <map>
+#include <sched.h>
+#include <vector>
+
+#include "umpire/util/Macros.hpp"
+#include "umpire/util/SICM_device.hpp"
+
+extern "C"
```

Comment:
Have you considered adding this to SICM directly?
```
#ifdef __cplusplus
extern "C" {
#endif
// ...
#ifdef __cplusplus
}
#endif
```

Reply:
I've made a pull request in SICM. I'll hold off changing the `#include`s here until the pull request has been merged.

Reply:
7c23a51849e8f2fd669786a39444138807cd154b